### ISSUE TITLE
In GO metadata, change all ORCIDs used in GO from HTTP to HTTPS

### ISSUE THREAD

**Issue by kltm** [geneontology/go-site#1708]
Currently, there is a mix of HTTP and HTTPS used for the scheme in URIs for user ORCIDs. After some discussion we have decided to 1) convert all of our user metadata concerning ORCIDs over to HTTPS and 2) ensure at the schema test level that no further HTTP additions can be made.

(As part of this, there will be related tickets about upgrading ORCID information in the noctua-models.)

**Comment by kltm** [comment]
Also see: https://github.com/geneontology/noctua-models/issues/194

**Comment by kltm** [comment]
To make sure that everybody is aware, as there might be synchronization issues as we go through this (although it's unlikely to be much worse than it is now anyways) tagging @vanaukenk @ukemi @cmungall @tmushayahama @lpalbou @sierra-moxon @dustine32 .

**Comment by kltm** [comment]
And @balhoff (sorry!), I'm remembering that users.yaml and groups.yaml get ontologized; i.e.: https://github.com/geneontology/pipeline/blob/fb7da0ea388409e4cda2fd71aa90e0003a2ed840/Jenkinsfile#L794 . I'm assuming this gets absorbed into some larger GO ontology load at some point.

From my reading then, after we get this HTTPS update into go-site, we'll want to get a `snapshot` out and then proceed with https://github.com/geneontology/noctua-models/issues/194, testing in `dev` and then to `master`.

@pgaudet I think we probably also should wait until this current `release` is done to just have less things moving around at the same time.

**Comment by pgaudet** [comment]
Should we go ahead with this now, since the release is done and hopefully there is a new one next week or so?

**Comment by kltm** [comment]
@pgaudet Good question. The last release was the 16th and we're at about a week away from the next release attempt. I think that if we assume that we'll be "late" or don't care about hitting the first attempt, we can go ahead and start give it a try.

**Comment by kltm** [comment]
@vanaukenk , thinking about how the upgraded data will roll around, if we want to do testing in noctua-dev, I think I would recommend either waiting to get started until after this Thursday's restart or getting started now and skipping this Thursday's restart.

**Comment by lpalbou** [comment]
Let me know when this passes through. I think it should be ok, but just to be on the safe side, I would like to test the GO-CAM API, Noctua Landing Page, ART and Noctua Form.

**Comment by ukemi** [comment]
Should MGI go ahead and change the upstream data?

**Comment by kltm** [comment]
For everybody following along, the data timeline would look like:

- [ ] update go-site user.yaml metadata
- [ ] allow this data to get out as part of a `snapshot` release (would appear in preassembled ontologies)
- [ ] update noctua-models `dev` branch models to new orchids (and then restart, picking up the new ontologies)
- [ ] update noctua-models `master` branch models to new orchids (and then restart, picking up the new ontologies)

**Comment by kltm** [comment]
@lpalbou Absolutely--sounds great. Which of the steps above are you most interested in?

**Comment by kltm** [comment]
@ukemi Probably not necessary, since we'll be doing everybody in bulk at the same time anyways.

**Comment by ukemi** [comment]
OK. So you are going to transform the upstream GPAD that @dustine32 imports from MGI?

**Comment by kltm** [comment]
@ukemi Sorry, I misinterpreted "upstream". As far as your MGI data goes, if you update it before we start, it will be out-of-sync until we make the changes. Practically speaking, since I believe the only place it expresses itself is in your export GPAD, and that is in production at no location, I believe it would be fine to do now. (Although I may have missed something--that's why this stuff is always so fun!)

**Comment by ukemi** [comment]
OK. We are forging ahead on the MGI end. I started a ticket. At this point the worst thing I see is that the curators' names won't resolve on the imports.

**Comment by lpalbou** [comment]
> @lpalbou Absolutely--sounds great. Which of the steps above are you most interested in?

Steps 3 and 4:
update of dev models = test NLP, ART, NF
update of prod models = GO-CAM API & GO website

**Comment by kltm** [comment]
I was planning on starting this today, but would like to wait until we have a recent good `snapshot` run complete (we lost a couple of runs to worm anatomy ontology issues and then some upstream ftp issues) so we don't keep people waiting as we debug.

**Comment by kltm** [comment]
A release is freshly out and noctua just cycled; I'm going to go ahead and start on this https://github.com/geneontology/go-site/issues/1708#issuecomment-866180607

**Comment by kltm** [comment]
@vanaukenk Now merged into go-site `master`, we should see results after the next `snapshot` success https://github.com/geneontology/go-site/issues/1708#issuecomment-866180607

**Comment by vanaukenk** [comment]
@kltm Just checking in with this one and confirming that the GO-CAM models metadata has not yet been updated on dev or production.  

@hdrabkin asked about the display of curator orcids on the production Noctua landing page.

Any curators whose orcid was originally https has their name displayed, while those whose orcids were originally http currently only show the http orcid (i.e. the original entry).  

![image](https://user-images.githubusercontent.com/1730534/125977048-e6b266d2-0102-4bf2-8133-183e0a68229b.png)

**Comment by kltm** [comment]
@vanaukenk The current state is here: https://github.com/geneontology/go-site/issues/1708#issuecomment-866180607
While the new updated is merged into go-site `master`, are still waiting on a successful `snapshot` before it starts percolating out to more places.

**Comment by kltm** [comment]
@vanaukenk noctua-dev should (I believe) be ready to test for HTTP -> HTTPS orcid upgrade.

https://github.com/geneontology/go-site/issues/1708#issuecomment-866180607

**Comment by kltm** [comment]
Notinghttps://github.com/geneontology/noctua/issues/730

**Comment by vanaukenk** [comment]
@kltm 
@ukemi made some changes to a model this morning and now both his name and http orcid are associated with the model and displaying on the Noctua landing page.
I'm not sure why this is; I updated a model last night and this didn't happen.

**Comment by ukemi** [comment]
Update. This problem seemed to go away after logging out and back in again. However, the two models that I modified before that still need to be fixed.

**Comment by kltm** [comment]
Okay, we'll need to coordinate a barista reset and an update for the models.
For the latter, we can possibly coordinate that with the update we'll need to be doing with the ZFIN models.

**Comment by kltm** [comment]
I believe this cleared w/o fanfare some time ago?
Please reopen if I'm mistaken.
